OpenChange/MAPIStoreAppointmentWrapper.m

```diff
@@ -1836,15 +1836,26 @@ - (void) _setupAlarm
   NSUInteger count, max;
   iCalAlarm *currentAlarm;
   NSString *action;
+  NSCalendarDate *now, *alarmDate;
 
   alarms = [event alarms];
   max = [alarms count];
+  now = [NSCalendarDate calendarDate];
   for (count = 0; !alarm && count < max; count++)
     {
       currentAlarm = [alarms objectAtIndex: count];
-      action = [[currentAlarm action] lowercaseString];
-      if (!action || [action isEqualToString: @"display"])
-        ASSIGN (alarm, currentAlarm);
+
+      // Only handle 'display' alarms
+      action = [currentAlarm action];
```

Comment:
With this patch, you miss the condition where `action` is `nil` (`!action`) to set up the alarm, or is this another fix?